02-microarray/pathway-analysis_microarray_02_ora.Rmd

```diff
@@ -0,0 +1,483 @@
+
+So, we will need to provide to `clusterProfiler` two genes lists: 
+
+1) Our genes of interest (`n`) 
+2) What genes were in our total background set (`N`). (All genes that originally had an opportunity to be measured). 
+
+## Determine our genes of interest list
+
+We will use our differential expression results to get a genes of interest list. 
+Let's use our adjusted p values as a cutoff 
+
+```{r}
+# Select genes that are below a cutoff
+genes_of_interest <- dge_annot_df %>%
+  # Here we want the top 100 fold changes for this group
+  dplyr::filter(adj.P.Val < 0.05) %>%
+  # We are extracting the gene symbols as a vector
+  dplyr::pull(gene_symbol)
+```
+
+There are a lot of ways we could make a genes of interest list, and this using a p-value cutoff is just one way you can do this; you could also pick a "top 100" or really use whatever you are basing your "interest" on. 
+
+ORA generally requires you make some sort of arbitrary decision to obtain your genes of interest list and this is one of the approach's weaknesses -- to get to a gene list we've removed all other context.
+
+Because one `gene_symbol` may map to multiple Ensembl IDs, we need to make sure we have no repeated gene symbols in this list. 
+
+```{r}
+# Reduce to only unique gene symbols
+genes_of_interest <- unique(as.character(genes_of_interest))
+
+# Let's print out some of these genes
+head(genes_of_interest)
+```
+
+## Determine our background set gene list
+
+Sometimes folks consider genes from the entire genome to comprise the background, but for our microarray data, we should consider all genes that were measured as our background set. 
+In other words, if we are unable to detect a gene, it should not be in our background set.
+
+We can obtain our detected genes list from our data frame, `dge_annot_df` (which we haven't done filtering on).
+
+```{r}
+background_set <- unique(as.character(dge_annot_df$gene_symbol))
+```
+
+## Run ORA using the `enricher()` function
+
+Now that we have our background set, our genes of interest, and our pathway information, we're ready to run ORA using the `enricher()` function.
+
+```{r}
+kegg_ora_results <- enricher(
+  gene = genes_of_interest, # A vector of your genes of interest
+  pvalueCutoff = 0.1, # Can choose a FDR cutoff
+  pAdjustMethod = "BH", # What method for multiple testing correction should we use
+  universe = background_set, # A vector containing your background set genes
+  # The pathway information should be a data frame with a term name or
+  # identifier and the gene identifiers
+  TERM2GENE = dplyr::select(
+    hs_kegg_df,
+    gs_name,
+    gene_symbol
+  )
+)
+```
+
+*Note: using `enrichKEGG()` is a shortcut for doing ORA using KEGG, but the approach we covered here can be used with any gene sets you'd like!*
```

Comment:
I would still say in this scenario (like training) we don't know what users are going to want so using the more general function and specifying `KEGG` still makes sense to me. We can replace this sentence with information on how to alter the general function to something else that they want.

Reply:
I think this comment needs to be evaluated in the context of the big picture. A long, general example may be unhelpful in the context of a product that is intended to get folks to an answer more quickly. Several examples where one of them is KEGG and one of them is for custom gene sets may be better.

Reply:
> product that is intended to get folks to an answer more quickly. 

I guess I've never thought of "quickly" as being a part of the goals of this project. 

> Several examples where one of them is KEGG and one of them is for custom gene sets may be better.

We will automatically have another ORA example in RNA-seq that we can use a different non-KEGG strategy there.
But sometimes we've used a strategy of here's us picking a method in this example but here's some information on how you'd alter this to switch to a different method. Do you think it's different enough that we wouldn't be able to send users to the docs and have some comments about how to switch to another method and that be sufficient? - I will evaluate this question myself too going through the examples, it just sounds like you think this should become an ORA series of sorts.

Reply:
> But sometimes we've used a strategy of here's us picking a method in this example but here's some information on how you'd alter this to switch to a different method. Do you think it's different enough that we wouldn't be able to send users to the docs and have some comments about how to switch to another method and that be sufficient?

I guess my concern with the state of this example is that it's long and needs more work or context around the choices being made. Put another way, too much information in some areas and not enough information in others. An example would be the chunk that starts at line 221 - that's included in training to specifically practice filtering. Here we could extract the KEGG stuff from `msigdbr` in a single step and kick the discussion about what gene sets are available to the introduction. I don't think we need to work all of that out now, because it will probably evolve as the rest of the examples come together. 

@jashapiro pointed out to me that highly specific examples could come off like explicit endorsements. One of our goals with these examples should be to get people doing what they want to be doing quickly, but we need to make it clear when we are making choices.

Reply:
> @jashapiro pointed out to me that highly specific examples could come off like explicit endorsements

*implicit* endorsement

Reply:
👆 that's what I meant

Reply:
Ah yes okay. This helps me know more about what the goal is. I will use this to try to rethink some of these steps and instructions.